pkg/splunk/splunk.go

```diff
@@ -0,0 +1,197 @@
+package splunk
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// SplunkHook provides a logrus hook which enables error logging to splunk platform.
+// This is helpful in order to provide better monitoring and alerting on errors
+// as well as the given error details can help to find the root cause of bugs.
+type Splunk struct {
+	levels        []logrus.Level
+	tags          map[string]string
+	splunkClient  piperhttp.Client
+	correlationID string
+	splunkDsn     string
+	splunkToken   string
+	splunkIndex   string
+
+	// boolean which forces to send all logs on error or none at all
+	sendLogs bool
+
+	// How big can be batch of messages
+	postMessagesBatchSize int
+}
+
+var SplunkClient *Splunk
+
+func Initialize(correlationID, dsn, token, index string, sendLogs bool) error {
+	log.Entry().Debugf("Initializing Splunk with DSN %v", dsn)
+
+	client := piperhttp.Client{}
+
+	client.SetOptions(piperhttp.ClientOptions{
+		MaxRequestDuration:        5 * time.Second,
+		Token:                     token,
+		TransportSkipVerification: true,
+	})
+
+	SplunkClient = &Splunk{
+		splunkClient:          client,
+		splunkDsn:             dsn,
+		splunkToken:           "Splunk " + token,
+		splunkIndex:           index,
+		correlationID:         correlationID,
+		postMessagesBatchSize: 1000,
+		sendLogs:              sendLogs,
+	}
```

Comment:
Do HTTP and Splunk client need the prefix?

```suggestion
      if !strings.HasPrefix(token, "Splunk ") {
          log.Entry.Debug("prefixing token")
          token = fmt.sprintf("Splunk %s", token)
      }
	client.SetOptions(piperhttp.ClientOptions{
		MaxRequestDuration:        5 * time.Second,
		Token:                     token,
		TransportSkipVerification: true,
	})

	SplunkClient = &Splunk{
		splunkClient:          client,
		splunkDsn:             dsn,
		splunkToken:          token,
		splunkIndex:           index,
		correlationID:         correlationID,
		postMessagesBatchSize: 1000,
		sendLogs:              sendLogs,
	}
```